README.md

```diff
@@ -1,18 +1,151 @@
+# but is included for completeness.
+[registry."acme.registry.com"]
+# This field is only required if more that one protocol is supported. It indicates which protocol
+# to use by default. If this is not set, then the fallback (oci) will be used.
+default = "warg"
+[registry."acme.registry.com".warg]
+# A path to a valid warg config file. If this is not set, the `wkg` CLI (but not the libraries) 
+# will attempt to load the config from the default location(s).
+config_file = "/a/path"
+[registry."acme.registry.com".oci]
+# The auth field can either be a username/password pair, or a base64 encoded `username:password` 
+# string. If no auth is set, the `wkg` CLI (but not the libraries) will also attempt to load the
+# credentials from the docker config.json. This field is also optional and if not set, anonymous
+# auth will be used. If you're just pulling from a public registry, this is likely not required.
+# If you're using a private registry and/or publishing, you'll almost certainly need to set this.
+auth = { username = "open", password = "sesame" }
+# This is an optional field that tells the OCI client to use a specific http protocol. If this is
+# not set or not one of the accepted values of "http" or "https", then the default (https) will
+# be used.
+protocol = "https"
+[registry."acme.registry.com".local]
+# This is a required field that specifies the root directory on a filesystem where the components
+# are stored. This is mostly used for local development and testing.
+root = "/a/path"
+
+# If a registry only has a config section for one protocol, then that protocol is automatically
+# the default. The following is equivalent to:
+# [registry."example.com"]
+# default = "warg"
+# [registry."example.com".warg]
+# config_file = "/a/path"
+[registry."example.com".warg]
+config_file = "/a/path"
+```
+
+### Well-known metadata
+
+The `wkg` tool and libraries expect a `registry.json` file to be present at a specific location to
+indicate to the tooling where the components are stored. If a registry was `example.com`, then the
+tooling will attempt to find a `registry.json` file at
+`https://example.com/.well-known/wasm-pkg/registry.json`. 
+
+For OCI registries, the JSON looks like this:
+
+```json
+{
+        "ociRegistry": "ghcr.io",
+        "ociNamespacePrefix": "webassembly/"
+}
+```
+
+The `ociRegistry` field is the base URL of the OCI registry, and the `ociNamespacePrefix` field is
+the prefix that is used to store components in the registry. So in the example above (which is for
+wasi.dev), the components will be available at `ghcr.io/webassembly/$NAMESPACE/$PACKAGE:$VERSION`
+(e.g. `ghcr.io/webassembly/wasi/http:0.2.1`).
+
+For Warg registries, the JSON looks like this:
+
+```json
+{
+  "wargUrl": "https://warg.wa.dev"
+}
+```
+
+The `wargUrl` field is the base URL of the Warg registry used when connecting the client.
+Namespacing for warg is built in to the protocol.
```

Comment:
I think is the old format with the "backward compatible support": https://github.com/bytecodealliance/wasm-pkg-tools/blob/main/crates/wasm-pkg-common/src/metadata.rs#L15-L32

https://github.com/bytecodealliance/wasm-pkg-tools/blob/main/crates/wasm-pkg-common/src/metadata.rs#L90-L98

```json
{
  "preferredProtocol":"warg",
  "warg": {"url":"https://warg.example.com"},
  "oci": {"registry": "oci.example.com", "namespacePrefix": "prefix/"}
}
```

Reply:
This has been updated